torchmetrics/image/map.py

```diff
@@ -0,0 +1,369 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import sys
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+import torch
+from torch import Tensor
+
+from torchmetrics.metric import Metric
+from torchmetrics.utilities.imports import _PYCOCOTOOLS_AVAILABLE
+
+if _PYCOCOTOOLS_AVAILABLE:
+    from pycocotools.coco import COCO
+    from pycocotools.cocoeval import COCOeval
+else:
+    COCO, COCOeval = None, None
+
+COCO_STATS_MAP_VALUE_INDEX = 0
+COCO_STATS_MAR_VALUE_INDEX = 8
+
+
+@dataclass
+class MAPMetricResults:
+    """Dataclass to wrap the final mAP results."""
+
+    map_value: Tensor
+    mar_value: Tensor
+    map_per_class_value: List[Tensor]
+    mar_per_class_value: List[Tensor]
+
+
+class _hide_prints:
+    """Internal helper context to suppress the default output of the pycocotools package."""
+
+    def __init__(self) -> None:
+        self._original_stdout = None
+
+    def __enter__(self) -> None:
+        self._original_stdout = sys.stdout  # type: ignore
+        sys.stdout = open(os.devnull, "w")
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:  # type: ignore
+        sys.stdout.close()
+        sys.stdout = self._original_stdout  # type: ignore
+
+
+def _input_validator(preds: List[Dict[str, torch.Tensor]], targets: List[Dict[str, torch.Tensor]]) -> None:
+    """Ensure the correct input format of `preds` and `targets`"""
+
+    if not isinstance(preds, list):
```

Comment:
I suggest checking that preds and targets are ``Sequence``s, instead of ``list``s, or ``(list, tuple)`` if ``Sequence`` is not available. Requiring that an input argument is a ``list`` can cause unnecessary trouble in Python.

Reply:
Done. I would leave exception and docstring as they are, fine for you?

Reply:
Fine.